sdk/messaging/azservicebus/receiver.go

```diff
@@ -372,6 +400,10 @@ func (r *Receiver) receiveMessagesImpl(ctx context.Context, maxMessages int, opt
 		return nil, internal.NewErrNonRetriable(fmt.Sprintf("maxMessages cannot exceed %d", r.maxAllowedCredits))
 	}
 
+	if msgs := r.receiveFromCache(maxMessages); len(msgs) > 0 {
```

Comment:
Can this be predicated to run only when the receiver is closed? I would have thought that we only need to do this post closing.

Reply:
It works out the same because the array of stored messages is only populated when the receiver is closed.